### DISCUSSION TITLE
I hope to place the @nx._dispatch decorator at the first position among method decorators.

### DISCUSSION THREAD

**Discussion by MikuSugar** [networkx/networkx#7205] [Q&A] [answered]
such as
```py
@py_random_state(5)
@nx._dispatch(edge_attrs="weight")
def betweenness_centrality(
    G, k=None, normalized=True, weight=None, endpoints=False, seed=None
):
```
For instance, when I call nx.betweenness_centrality(nx_graph, normalized=False), the py_random_state within it generates a 'seed' parameter. However, the backend I'm currently developing doesn't support the 'seed' parameter. Placing the nx._dispatch decorator before py_random_state doesn't affect the backend I'm developing.

**Comment by MikuSugar** [comment]
Perhaps I could submit a PR to place all nx._dispatch decorators at the beginning?

**Comment by eriknw** [comment]
It's okay to simply ignore the seed parameter. For example, the cugraph backend occasionally does this, and it's reflected in the docs such as here (scroll to the bottom): https://networkx.org/documentation/stable/reference/algorithms/generated/networkx.algorithms.community.louvain.louvain_communities.html

**Reply by MikuSugar** [reply]
Thank you for your response, I'm doing it this way now too.
I think it's more suitable for the backend to have backend decorators placed first, followed by parameter-related decorators, allowing the backend to handle parameter-related tasks on its own.

**Reply by dschult** [reply, marked as answer]
The seed parameter decorator is not the only one that can be affected by this. For example, `@number_or_nodes` converts the input. I guess the question is whether backends are expected to interpret the networkx function signature as documented in networkx, or do we give them the freedom to parse the input the way they want to and if they want to use the networkx api they will have to do some input parsing that would normally be done for them by networkx (in the same way that networkx would do it). 

My initial reaction is that letting backends handle the raw function input will lead to more trouble than help. We're trying to provide a unified function signature for all backends, and that becomes compromised if backends have to process the inputs. Of course, bckends can process them any way they want -- including undoing what the networkx decorator does. We are only choosing the default/easy to handle behavior in this choice -- not enforcing much of anything.

**Reply by MikuSugar** [reply]
@dschult Thank you for your reply. After careful consideration of your words, you're right. A unified function signature and parameter handling are indeed better, reducing the learning curve for users.
